soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala

```diff
@@ -0,0 +1,527 @@
+package com.socrata.soql.codecs
+
+import com.rojoma.json.v3.ast._
+import com.rojoma.json.v3.codec.{JsonDecode, JsonEncode}
+import com.rojoma.json.v3.util.JsonUtil
+import com.vividsolutions.jts.geom._
+
+import scala.util.Try
+import java.io.IOException
+
+import com.socrata.soql.types._
+import com.socrata.soql.types.obfuscation.CryptProvider
+
+import com.socrata.thirdparty.geojson.JtsCodecs.geoCodec
+
+trait JsonColumnCommonRep {
+  val representedType: SoQLType
+}
+
+trait JsonColumnReadRep extends JsonColumnCommonRep {
+  def fromJValue(input: JValue): Option[SoQLValue]
+}
+
+trait JsonColumnWriteRep extends JsonColumnCommonRep {
+  def toJValue(value: SoQLValue): JValue
+  protected def stdBadValue: Nothing =
+    sys.error("Incorrect value passed to toJValue")
+}
+
+trait JsonColumnRep extends JsonColumnReadRep with JsonColumnWriteRep
+
+class CodecBasedJsonColumnRep[TrueCV: JsonEncode: JsonDecode](
+    val representedType: SoQLType,
+    unwrapper: SoQLValue => TrueCV,
+    wrapper: TrueCV => SoQLValue
+) extends JsonColumnRep {
+  def fromJValue(input: JValue) =
+    if (JNull == input) Some(SoQLNull)
+    else JsonDecode[TrueCV].decode(input).toOption.map(wrapper)
+
+  def toJValue(input: SoQLValue) =
+    if (SoQLNull == input) JNull
+    else JsonEncode[TrueCV].encode(unwrapper(input))
+}
+
+object JsonColumnRep {
+  // this is used to en/decrypt row IDs and values.  The key DOESN'T MATTER,
+  // because this system doesn't care about the actual values in a SoQLID or
+  // a SoQLValue; it just cares that it can recognize and reproduce them.
+  //
+  // It would be better to not care about the decrypted values at all, but alas
+  // that is not how SoQLID and SoQLVersion work.
+  private[this] val cryptProvider = new CryptProvider(Array[Byte](0))
+
+  val IdClearNumberRep = new SoQLID.ClearNumberRep(cryptProvider)
```

Comment:
ClearNumberRep likewise isn't a thing that something on the non-SF path should care about